src/libraries/System.Private.CoreLib/src/System/Reflection/FieldAccessor.cs

```diff
@@ -0,0 +1,534 @@
+                    case FieldAccessorType.InstanceReferenceType:
+                        VerifyTarget(obj);
+                        Debug.Assert(obj != null);
+                        return Volatile.Read(ref Unsafe.As<byte, object>(ref Unsafe.AddByteOffset(ref obj.GetRawData(), _addressOrOffset)));
+
+                    case FieldAccessorType.InstanceValueType:
+                    case FieldAccessorType.InstanceValueTypeSize1:
+                    case FieldAccessorType.InstanceValueTypeSize2:
+                    case FieldAccessorType.InstanceValueTypeSize4:
+                    case FieldAccessorType.InstanceValueTypeSize8:
+                        VerifyTarget(obj);
+                        Debug.Assert(obj != null);
+                        return RuntimeHelpers.Box(
+                            _methodTable,
+                            ref Unsafe.AddByteOffset(ref obj.GetRawData(), _addressOrOffset));
+
+                    case FieldAccessorType.InstancePointerType:
+                        VerifyTarget(obj);
+                        Debug.Assert(obj != null);
+                        return Pointer.Box(
+                            (void*)Unsafe.As<byte, IntPtr>(ref Unsafe.AddByteOffset(ref obj.GetRawData(), _addressOrOffset)),
+                            _fieldInfo.FieldType);
+
+                    case FieldAccessorType.StaticReferenceType:
+                        return Volatile.Read(ref Unsafe.As<IntPtr, object>(ref *(IntPtr*)_addressOrOffset));
+
+                    case FieldAccessorType.StaticValueType:
+                    case FieldAccessorType.StaticValueTypeSize1:
+                    case FieldAccessorType.StaticValueTypeSize2:
+                    case FieldAccessorType.StaticValueTypeSize4:
+                    case FieldAccessorType.StaticValueTypeSize8:
+                        return RuntimeHelpers.Box(_methodTable, ref Unsafe.AsRef<byte>(_addressOrOffset.ToPointer()));
+
+                    case FieldAccessorType.StaticValueTypeBoxed:
+                        // Re-box the value.
+                        return RuntimeHelpers.Box(
+                            _methodTable,
+                            ref Unsafe.As<IntPtr, object>(ref *(IntPtr*)_addressOrOffset).GetRawData());
+
+                    case FieldAccessorType.StaticPointerType:
+                        return Pointer.Box((void*)Unsafe.As<byte, IntPtr>(
+                            ref Unsafe.AsRef<byte>(_addressOrOffset.ToPointer())), _fieldInfo.FieldType);
+
+                    case FieldAccessorType.SlowPathUntilClassInitialized:
+                        if (!IsStatic())
+                        {
+                            VerifyTarget(obj);
+                        }
+
+                        domainInitialized = false;
+                        object? ret = RuntimeFieldHandle.GetValue(_fieldInfo, obj, (RuntimeType)_fieldInfo.FieldType, _fieldInfo.m_declaringType, ref domainInitialized);
+                        if (domainInitialized)
```

Comment:
Can `domainInitialized` ever be false here?

Reply:
No; removed that. Thanks. Not sure why it was originally done that way - perhaps at one point something to do with helping to support setting initonly fields with reflection but only during a class constructor.